common/lib/xmodule/xmodule/js/src/video/01_initialize.js

```diff
@@ -523,67 +556,40 @@ function (VideoPlayer, i18n, moment) {
-                        );
-
-                        // When the youtube link doesn't work for any reason
-                        // (for example, the great firewall in china) any
-                        // alternate sources should automatically play.
-                        if (!_prepareHTML5Video(self)) {
-                            console.log(
-                                '[Video info]: Continue loading ' +
-                                'YouTube video.'
-                            );
-
-                            // Non-YouTube sources were not found either.
-
-                            el.find('.video-player div')
-                                .removeClass('hidden');
-                            el.find('.video-player h3')
-                                .addClass('hidden');
-
-                            // If in reality the timeout was to short, try to
-                            // continue loading the YouTube video anyways.
-                            self.fetchMetadata();
-                            self.parseSpeed();
-                        } else {
-                            console.log(
-                                '[Video info]: Change player mode to HTML5.'
-                            );
+            console.log('[Video info]: YouTube API is not loaded. Will try to load...');
+            window.setTimeout(function () {
+                // If YouTube API will load OK, it will run `onYouTubeIframeAPIReady`
+                // callback, which will set `state.youtubeApiAvailable` to `true`.
+                // If something goes wrong at this stage, `state.youtubeApiAvailable` is
+                // `false`.
+                if (!self.youtubeIsAvailable) {
+                    console.log('[Video info]: YouTube API is not available.');
+                }
+                self.el.trigger('youtube_availability', [self.youtubeIsAvailable]);
+            }, self.config.ytTestTimeout);
 
-                            // In-browser HTML5 player does not support quality
-                            // control.
-                            el.find('a.quality_control').hide();
-                        }
-                    } else {
-                        console.log(
-                            '[Video info]: Start player in YouTube mode.'
-                        );
+            var scriptTag = document.createElement('script');
+            scriptTag.src = document.location.protocol + '//' + this.config.ytApiUrl;
 
-                        self.fetchMetadata();
-                        self.parseSpeed();
-                    }
+            $(scriptTag).on('load', function() {
+                self.loadYoutubePlayer();
+            });
+            $(scriptTag).on('error', function() {
+                self.loadHtmlPlayer();
+            });
 
-                    _setConfigurations(self);
-                    _renderElements(self);
-                });
+            if(!window.injectScriptTagToDOM) {
```

Comment:
This can go into `methodsDict`. And then in `10_main.js` line 162, you can do:
```
window.Video.loadYouTubeIFrameAPI = initialize.prototype.loadYouTubeIFrameAPI
```